EventFilter/TotemRawToDigi/src/RawToDigiConverter.cc

```diff
@@ -0,0 +1,271 @@
+  testECMostFrequent(conf.getParameter<unsigned int>("testECMostFrequent")),
+  testBCMostFrequent(conf.getParameter<unsigned int>("testBCMostFrequent")),
+  
+  EC_min(conf.getUntrackedParameter<unsigned int>("EC_min", 10)),
+  BC_min(conf.getUntrackedParameter<unsigned int>("BC_min", 10)),
+  
+  EC_fraction(conf.getUntrackedParameter<double>("EC_fraction", 0.6)),
+  BC_fraction(conf.getUntrackedParameter<double>("BC_fraction", 0.6))
+{
+}
+
+//----------------------------------------------------------------------------------------------------
+
+void RawToDigiConverter::RunCommon(const VFATFrameCollection &input, const TotemDAQMapping &mapping,
+      map<TotemFramePosition, TotemVFATStatus> &status)
+{
+  // map which will contain FramePositions from mapping missing in raw event
+  map<TotemFramePosition, TotemVFATInfo> missingFrames(mapping.VFATMapping);
+
+  // EC and BC checks (wrt. the most frequent value), BC checks per subsystem
+  CounterChecker ECChecker(CounterChecker::ECChecker, "EC", EC_min, EC_fraction, verbosity);
+  CounterChecker BCChecker(CounterChecker::BCChecker, "BC", BC_min, BC_fraction, verbosity);
+
+  // loop over data frames
+  stringstream ees;
+
+  for (VFATFrameCollection::Iterator fr(&input); !fr.IsEnd(); fr.Next())
+  {
+    stringstream fes;
+    bool stopProcessing = false;
+    
+    // skip data frames not listed in the DAQ mapping
+    auto mappingIter = mapping.VFATMapping.find(fr.Position());
+    if (mappingIter == mapping.VFATMapping.end())
+    {
+      unknownSummary[fr.Position()]++;
+      continue;
+    }
+
+    // contain information about processed frame
+    TotemVFATStatus &actualStatus = status[fr.Position()];
+
+    // remove from missingFrames
+    auto iter = missingFrames.find(fr.Position());
+    missingFrames.erase(iter);
+
+    // check footprint
+    if (testFootprint != tfNoTest && !fr.Data()->checkFootprint())
+    {
+      fes << "    invalid footprint\n";
+      if ((testFootprint == tfErr))
+      {
+        actualStatus.setFootprintError();
+        stopProcessing = true;
+      }
+    }
+    
+    // check CRC
+    if (testCRC != tfNoTest && !fr.Data()->checkCRC())
+    {
+      fes << "    CRC failure\n";
+      if (testCRC == tfErr)
+      {
+        actualStatus.setCRCError();
+        stopProcessing = true;
+      }
+    }
+
+    // check the id mismatch
+    if (testID != tfNoTest && fr.Data()->isIDPresent() && (fr.Data()->getChipID() & 0xFFF) != (mappingIter->second.hwID & 0xFFF))
+    {
+      fes << "    ID mismatch (data: 0x" << hex << fr.Data()->getChipID()
+        << ", mapping: 0x" << mappingIter->second.hwID  << dec << ", symbId: " << mappingIter->second.symbolicID.symbolicID << ")\n";
+      if (testID == tfErr)
+      {
+        actualStatus.setIDMismatch();
+        stopProcessing = true;
+      }
+    }
+
+    // if there were errors, put the information to ees buffer
+    if (verbosity > 0 && !fes.rdbuf()->str().empty())
```

Comment:
count issues in some other way, not from an accumulation of stringstream buffer.
Depending on the noise level I could see this can take up significant memory and CPU resources.
I can suggest a simple struct { int errCount; stringstream es; } and in the code, append to es only if verbosity is non-zero ..